src/journal/journald-context.c

```diff
@@ -60,7 +61,34 @@
 /* Keep at most 16K entries in the cache. (Note though that this limit may be violated if enough streams pin entries in
  * the cache, in which case we *do* permit this limit to be breached. That's safe however, as the number of stream
  * clients itself is limited.) */
-#define CACHE_MAX (16*1024)
+#define CACHE_MAX_FALLBACK 128U
+#define CACHE_MAX_MAX (16*1024U)
+#define CACHE_MAX_MIN 64U
+
+static unsigned cache_max(void) {
+        static int cached = -1;
```

Comment:
Cache size depends on total memory size of the system, but that can change. Memory hot-plug is quite common in VMs.

I figure this can be fixed later, it at all.